engine/src/main/java/org/camunda/bpm/engine/impl/cmd/SaveAuthorizationCmd.java

```diff
@@ -50,6 +56,10 @@ public Authorization execute(CommandContext commandContext) {
 
     authorizationManager.validateResourceCompatibility(authorization);
 
+    if (isHistoricTaskInstanceResource()) {
+      provideRemovalTime(() -> getHistoricTaskInstance(commandContext));
```

Comment:
Why does this use a supplier as the method parameter? The first thing the method does is call `#get`.

Reply:
This is to resuse the `provideRemovalTime` method when tackling the Historic Process Instance feature.

Reply:
Not sure I can follow. Couldn't the historic process instance feature also just pass a HistoryEvent to the method?

Reply:
You are right! Might be a left over from an intermediate implementation state.